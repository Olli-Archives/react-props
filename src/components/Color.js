import React from 'react';
import PropTypes from 'prop-types';

export default function Color({ name, hex, red, green, blue }){
  return (
    <ul>
      <li>{name}</li>
      <li>{hex}</li>
      <li>red:{red}</li>
      <li>green:{green}</li>
      <li>blue:{blue}</li>
    </ul>
   
  );
}

Color.propTypes = {
  name: PropTypes.string.isRequired,
  hex: PropTypes.string.isRequired,
  red: PropTypes.number.isRequired,
  green: PropTypes.number.isRequired,
  blue: PropTypes.number.isRequired
}
;

