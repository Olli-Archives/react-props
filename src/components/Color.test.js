import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('test color component', ()=>{
  it('creates color html', ()=>{
    const wrapper = shallow(<Color name='blue' hex="#8673A1" red={55} green={44} blue={674} />);
    expect(wrapper).toMatchSnapshot();
  });
});
