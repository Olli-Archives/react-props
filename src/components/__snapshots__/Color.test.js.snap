// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test color component creates color html 1`] = `
<ul>
  <li>
    blue
  </li>
  <li>
    #8673A1
  </li>
  <li>
    red:
    55
  </li>
  <li>
    green:
    44
  </li>
  <li>
    blue:
    674
  </li>
</ul>
`;
